src/java/org/apache/cassandra/index/sai/disk/v1/sortedterms/SortedTermsReader.java

```diff
@@ -112,123 +108,259 @@ public SortedTermsReader(@Nonnull FileHandle termsDataFileHandle,
          */
-        public long pointId()
+        public @Nonnull ByteComparable seekForwardToPointId(long nextPointId)
         {
-            return pointId;
-        }
+            if (nextPointId < 0 || nextPointId > meta.termCount)
+                throw new IndexOutOfBoundsException(String.format("The target point id [%s] cannot be less than 0 or " +
+                                                                  "greater than the term count [%s]", nextPointId, meta.termCount));
+            assert nextPointId >= currentPointId : "Attempt to seek backwards in seekForwardsToPointId. Next pointId was "
+                                                   + nextPointId + " while current pointId is " + currentPointId;
+            if (nextPointId != currentPointId)
+            {
+                long blockIndex = nextPointId >>> blockShift;
+                if (blockIndex != currentBlockIndex)
+                {
+                    currentBlockIndex = blockIndex;
+                    resetPosition();
+                }
+            }
+            while (currentPointId < nextPointId)
+            {
+                readTerm(++currentPointId, currentTerm);
+                currentBlockIndex = currentPointId >>> blockShift;
+            }
 
-        /**
-         * Returns the current term data as {@link ByteComparable} referencing the internal term buffer.
-         * The term data stored behind that reference is valid only until the next call to
-         * {@link #advance} or {@link #seekToPointId(long)}.
-         */
-        public @Nonnull ByteComparable term()
-        {
             return ByteComparable.fixedLength(currentTerm.bytes, currentTerm.offset, currentTerm.length);
         }
 
         /**
-         * Positions the cursor on the target point id and reads the term at target to the current term buffer.
-         * <p>
-         * It is allowed to position the cursor before the first item or after the last item;
-         * in these cases the internal buffer is cleared.
+         * Finds the pointId for a term within a range of pointIds. The start and end of the range must not
+         * exceed the number of terms available.
          * <p>
-         * This method has constant complexity.
-         *
-         * @param pointId point id to lookup
-         * @throws IOException if a seek and read from the terms file fails
-         * @throws IndexOutOfBoundsException if the target point id is less than -1 or greater than the number of terms
+         * If the term is not in the block containing the start of the range a binary search is done to find
+         * the block containing the search. That block is then searched to return the pointId that corresponds
+         * to the term that either equal to or next highest to the term.
          */
-        public void seekToPointId(long pointId) throws IOException
+        public long partitionedSeekToTerm(ByteComparable term, long startingPointId, long endingPointId)
```

Comment:
Is `endingPointId` exclusive? Would perhaps mention that in the method JavaDoc...